src/publicTypes.ts

```diff
@@ -136,4 +136,56 @@ export type NamehashOptions = {
 };
 
 export const NamehashOptionsDefault = {format: 'hex', prefix: true} as const;
-export type CryptoRecords = Record<string, string | undefined>;
+
+export enum DnsRecordType {
+A="A",
+AAAA="AAAA",
+AFSDB="AFSDB",
+APL="APL",
+CAA="CAA",
+CDNSKEY="CDNSKEY",
+CDS="CDS",
+CERT="CERT",
+CNAME="CNAME",
+CSYNC="CSYNC",
+DHCID="DHCID",
+DLV="DLV",
+DNAME="DNAME",
+DNSKEY="DNSKEY",
+DS="DS",
+EUI48="EUI48",
+EUI64="EUI64",
+HINFO="HINFO",
+HIP="HIP",
+HTTPS="HTTPS",
+IPSECKEY="IPSECKEY",
+KEY="KEY",
+KX="KX",
+LOC="LOC",
+MX="MX",
+NAPTR="NAPTR",
+NS="NS",
+NSEC="NSEC",
+NSEC3="NSEC3",
+NSEC3PARAM="NSEC3PARAM",
+OPENPGPKEY="OPENPGPKEY",
+PTR="PTR",
+RP="RP",
+RRSIG="RRSIG",
+SIG="SIG",
+SMIMEA="SMIMEA",
+SOA="SOA",
+SRV="SRV",
+SSHFP="SSHFP",
+SVCB="SVCB",
+TA="TA",
+TKEY="TKEY",
+TLSA="TLSA",
+TSIG="TSIG",
+TXT="TXT",
+URI="URI",
+ZONEMD="ZONEMD",
+}
+
+export type DnsRecord = {type: DnsRecordType, ttl: number, value: string}
+export type CryptoRecords = Record<string, number | string | string[] | undefined>;
```

Comment:
How can I get an array of values from record "dns.A" in that case? Also on our docs, it was said that dns.A.ttl and all other ttl type records are holding a number.  If with numbers it is easy, they can be represented as a string what with arrays and many values?

Reply:
The case is that on the contract level all of them stored as strings so when we say `CryptoRecords` - no matter what format a record has - it is stored as string on a contract level. So we want to keep `CryptoRecords` type value bound to `string | undefined` because this is how the data is stored on the blockcchain. On other end `DnsRecord` is a type on top of basic `CryptoRecords` and it can be semantic.